### DISCUSSION TITLE
How can I set proxy for the built-in tools

### DISCUSSION THREAD

**Discussion by ddrcrow** [langgenius/dify#7833] [General]
### Self Checks

- [X] I have searched for existing issues [search for existing issues](https://github.com/langgenius/dify/issues), including closed ones.
- [X] I confirm that I am using English to submit this report (我已阅读并同意 [Language Policy](https://github.com/langgenius/dify/issues/1542)).
- [X] [FOR CHINESE USERS] 请务必使用英文提交 Issue，否则会被关闭。谢谢！:）
- [X] Please do not modify this template :) and fill in all the required fields.

### Content

When I used DuckDuckGo Search Tool, it reported:
```
Run failed: Node DuckDuckGo Search run failed: Failed to invoke tool: https://duckduckgo.com Exception: Error in request: error sending request for url (https://duckduckgo.com/): operation timed out
```
I suspect that it was blocked by firewall, do you know how to set a proxy for such built-in tools

**Comment by ddrcrow** [comment]
anyone can help me? thanks

**Comment by zhujinle** [comment]
Have you set the `HTTP_PROXY` and `HTTPS_PROXY` in `env` ?

FYI: [Dify's Doc](https://docs.dify.ai/v/zh-hans/getting-started/install-self-hosted/environments)

**Reply by ddrcrow** [reply]
I just saw SSRF_PROXY_HTTP_URL in .env, was not aware of adding such variables...
I will try to set HTTP_PROXY and HTTPS_PRXOY explictly in .env, will update you. many thanks

**Reply by zhujinle** [reply]
Sorry, I typed an extra symbol.
You should set these conf in `docker-compose.yaml` if you are running in Docker mode.
Our solution is to use a soft router as a network proxy, so we may not be able to provide further suggestions, this is for reference only.
FYI: issues #5669

**Reply by xueshunlin** [reply]
> I just saw SSRF_PROXY_HTTP_URL in .env, was not aware of adding such variables... I will try to set HTTP_PROXY and HTTPS_PRXOY explictly in .env, will update you. many thanks

Hello wonder does this work? thanks
